cura/UI/WhatsNewPagesModel.py

```diff
@@ -1,22 +1,97 @@
-# Copyright (c) 2019 Ultimaker B.V.
+# Copyright (c) 2021 Ultimaker B.V.
 # Cura is released under the terms of the LGPLv3 or higher.
-
 from .WelcomePagesModel import WelcomePagesModel
 
+import os
+from typing import Optional, Dict, List, Tuple
+from PyQt5.QtCore import pyqtProperty, pyqtSlot
+from UM.Logger import Logger
+from UM.Resources import Resources
 
 #
 # This Qt ListModel is more or less the same the WelcomePagesModel, except that this model is only for showing the
 # "what's new" page. This is also used in the "Help" menu to show the changes log.
 #
 class WhatsNewPagesModel(WelcomePagesModel):
 
+    image_formats = [".png", ".jpg", ".jpeg", ".gif", ".svg"]
+    text_formats = [".txt", ".htm", ".html"]
+    image_key = "image"
+    text_key = "text"
+
+    @classmethod
+    def _collectOrdinalFiles(cls, resource_type: int, include: List[str]) -> Tuple[Dict[int, str], int]:
+        result = {}  # Dict[int, str]
+        highest = -1
+        try:
+            folder_path = Resources.getPath(resource_type, "whats_new")
+            for _, _, files in os.walk(folder_path):
+                for filename in files:
+                    basename = os.path.basename(filename)
+                    base, ext = os.path.splitext(basename)
+                    if ext not in include or not base.isdigit():
+                        continue
+                    page_no = int(base)
+                    highest = max(highest, page_no)
+                    result[page_no] = os.path.join(folder_path, filename)
+        except FileNotFoundError:
+            Logger.logException("w", "Could not find 'whats_new' folder for resource-type {0}".format(resource_type))
+        return result, highest
+
+    @classmethod
+    def _loadText(cls, filename: str) -> str:
```

Comment:
Same goes for this one. Shouldn't this be static?

Reply:
Also.